baseapp/genesis.go

```diff
@@ -0,0 +1,22 @@
+package baseapp
+
+import (
+	"errors"
+
+	"cosmossdk.io/core/genesis"
+	"github.com/cometbft/cometbft/abci/types"
+)
+
+var _ genesis.TxHandler = (*BaseApp)(nil)
+
+// ExecuteGenesisTx implements genesis.GenesisState from
+// cosmossdk.io/core/genesis to set initial state in genesis
+func (ba BaseApp) ExecuteGenesisTx(tx []byte) error {
+	res := ba.DeliverTx(types.RequestDeliverTx{Tx: tx})
```

Comment:
Why is there still a `DeliverTx` method and `RequestDeliverTx` type?

Reply:
this is against main. in a follow up pr to comet i will amend to be the private method

Reply:
I see. ACK